From 55502932e95226283a6ce064ed0aa49b50b244bf Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Thu, 2 Nov 2023 13:12:55 +0000
Subject: [PATCH 0704/1085] dts: rp1: Add spi6, fix spi1 #address-cells

spi6 won't be useful on Pi 5 because it can't be enabled on the 40-pin
header, but include it for completeness.

Also fix the #address-cells value for spi1, otherwise the kernel will
reject attempts to apply the, say, spi1-2cs overlay at runtime.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/broadcom/rp1.dtsi | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

--- a/arch/arm/boot/dts/broadcom/rp1.dtsi
+++ b/arch/arm/boot/dts/broadcom/rp1.dtsi
@@ -187,7 +187,7 @@
 			interrupts = <RP1_INT_SPI1 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&rp1_clocks RP1_CLK_SYS>;
 			clock-names = "ssi_clk";
-			#address-cells = <0>;
+			#address-cells = <1>;
 			#size-cells = <0>;
 			num-cs = <2>;
 			dmas = <&rp1_dma RP1_DMA_SPI1_TX>,
@@ -262,6 +262,21 @@
 			dma-names = "tx", "rx";
 			status = "disabled";
 		};
+
+		rp1_spi6: spi@68000 {
+			reg = <0xc0 0x40068000  0x0 0x130>;
+			compatible = "snps,dw-apb-ssi";
+			interrupts = <RP1_INT_SPI6 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&rp1_clocks RP1_CLK_SYS>;
+			clock-names = "ssi_clk";
+			#address-cells = <1>;
+			#size-cells = <0>;
+			num-cs = <2>;
+			dmas = <&rp1_dma RP1_DMA_SPI6_TX>,
+			       <&rp1_dma RP1_DMA_SPI6_RX>;
+			dma-names = "tx", "rx";
+			status = "disabled";
+		};
 
 		// SPI7 is a target/slave interface
 		rp1_spi7: spi@6c000 {
